src/nwb_conversion_tools/tools/data_transfers.py

```diff
@@ -61,14 +63,143 @@ def get_globus_dataset_content_sizes(
 
     recursive_flag = " --recursive" if recursive else ""
     contents = json.loads(
-        _deploy_process(
+        deploy_process(
             command=f"globus ls -Fjson {globus_endpoint_id}:{path}{recursive_flag}", catch_output=True, timeout=timeout
         )
     )
     files_and_sizes = {item["name"]: item["size"] for item in contents["DATA"] if item["type"] == "file"}
     return files_and_sizes
 
 
+def transfer_globus_content(
+    source_id: str,
+    source_files: Union[str, List[List[str]]],
+    destination_id: str,
+    destination_folder: FolderPathType,
+    display_progress: bool = True,
+    progress_update_rate: float = 60.0,
+    progress_update_timeout: float = 600.0,
+) -> (bool, List[str]):
+    """
+    Track download progress for transferring content from source_id to destination_id:destination_folder.
+
+    Parameters
+    ----------
+    source_id : string
+        Source Globus ID.
+    source_files : string, or list of strings, or list of lists of strings
+        A string path or list-of-lists of string paths of files to transfer from the source_id.
+        If using a nested list, the outer level indicates which requests will be batched together.
+        If using a nested list, all items in a single batch level must be from the same common directory.
+
+        It is recommended to transfer the largest file(s) with minimal batching,
+        and to batch a large number of very small files together.
+
+        It is also generally recommended to submit up to 3 simultaneous transfer,
+        *i.e.*, `source_files` is recommended to have at most 3 items all of similar total byte size.
+    destination_id : string
+        Destination Globus ID.
+    destination_folder : FolderPathType
+        Absolute path to a local folder where all content will be transfered to.
+    display_progress : bool, optional
+        Whether or not to display the transfer as progress bars using `tqdm`.
+        Defaults to True.
+    progress_update_rate : float, optional
+        How frequently (in seconds) to update the progress bar display tracking the data transfer.
+        Defaults to 30 seconds.
+    progress_update_tiemout : float, optional
+        Maximum amount of time to monitor the transfer progress.
+        You may wish to set this to be longer when transferring very large files.
+        Defaults to 10 minutes.
+
+    Returns
+    -------
+    success : bool
+        If 'display_progress'=False, and the transfer was succesfully initiated, then this function returns True.
+
+        If 'display_progress'=True (the default), then this function returns the total status of all transfers
```

Comment:
```suggestion
        If 'display_progress'=True (the default), then this function returns the total size of all transfers
```
Is this what you mean?

Reply:
Maybe just 'progress' if 'status' is too equivocal with being boolean.

Reply:
No, 'status' is correct, everything here is boolean. `success` is the global status, `all(all_status)`, and `all_status` is the list of booleans.

Reply:
I see, so maybe: then this functions returns success=True if all transfer have succeeded?
I think what confuses me is the phrase "returns total status" by which you mean returns `all(all_status)`

Reply:
> I see, so maybe: then this functions returns success=True if all transfer have succeeded?

Correct.

> I think what confuses me is the phrase "returns total status" by which you mean returns all(all_status)

Yeah, 'total status' = combined success of all batches